### PR TITLE
Yevdokimov Danil | Attention, wanted in Arcane!

### PR DESCRIPTION
**Pull request by akabynda**
Did it!

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -1,3 +0,0 @@
-# Default ignored files
```
Comment (SpirinEgor) [inline .idea/.gitignore:1]: Delete these files in the main branch and fetch changes here. So they don't appear in PR.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿# Внимание, розыск в Аркейн!
+
+Всегда хотели узнать как выглядят подозреваемые в совершении преступление в стиле мультсериала Arcane? Просто запустите приложение и вставьте ссылку на подозреваемого с сайта https://moscow.sledcom.ru/attention/Podozrevaemie_v_sovershenii_prestuplenij.
+Приметы подозреваемого в его анкете должны быть указаны явно. Пример: "**Приметы на момент исчезновения:** рост 170-175 см, телосложение среднее, волосы темные, короткие, имеются лобные залысины, лицо овальное, нос большой с горбинкой."
+
+
+# Технические аспекты
+
+Не забудьте сделать pip install -r requirements.txt
```
Comment (SpirinEgor) [inline README.md:9]: But you don't have a `requirements.txt` file in PR and repo at all.

File: main.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
+import validators
+import streamlit as st
+import torch
+from diffusers import StableDiffusionPipeline
+import requests
+from bs4 import BeautifulSoup
+from deep_translator import GoogleTranslator
+
+model_id = "nitrosocke/Arcane-Diffusion"
```
Comment (SpirinEgor) [inline main.py]: Conventionally, it is better to use UPPER_CASE for constants.
```python
MODEL_ID = "nitrosocke/Arcane-Diffusion"
```

File: main.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+
+@st.cache(suppress_st_warning=True, allow_output_mutation=True)
+def init_pipe():
+    # Лаконично и понятно, думаю менять не стоит
```
Comment (SpirinEgor) [inline main.py]: If you have something to say for me or anyone else, it should be done in discussion or in chat. 😅
Comments in the code are for clarification, not for discussion.

**Review by SpirinEgor — CHANGES_REQUESTED** [review]
For now, your code is a mix of global statements and functions. Let's organize it a little bit and make explicit the main method. This main method will init all states and define app structure. You can call it under `if __name__ == "__main__"` statement.

File: main.py (hunk 4 above)
Comment (akabynda) [reply main.py]: Okay, I'll learn.

File: .idea/.gitignore (hunk 1 above)
Comment (akabynda) [reply .idea/.gitignore:1]: Corrected

File: main.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            pipe = StableDiffusionPipeline.from_pretrained(MODEL_ID)
+            pipe = pipe.to("cpu")
+    else:
+        pipe = StableDiffusionPipeline.from_pretrained(MODEL_ID)
+        pipe = pipe.to("cpu")
+    pipe.safety_checker = lambda images, clip_input: (images, False)
+    return pipe
+
+
+PIPE = init_pipe()
```
Comment (SpirinEgor) [inline main.py]: Why do you make it global if you cached it? You can call the function every time you need your pipeline.

File: main.py [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+
+def generate_arcane(text, num_inference_steps=50, guidance_scale=7.5):
+    return PIPE(prompt="arcane style, " + text, num_inference_steps=num_inference_steps,
```
Comment (SpirinEgor) [inline main.py]: You need a global variable only to use it in this function. That is strange 🤷‍♂️

**Review by SpirinEgor — COMMENTED** [review]
Get rid of global pipe and then all will be okay!

File: main.py (hunk 5 above)
Comment (akabynda) [reply main.py]: You are right, I have corrected the error.

**Pull request merged**